packages/core/src/components/StepList/StepLink.jsx

```diff
@@ -0,0 +1,39 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+export class StepLink extends React.PureComponent {
+  handleClick(event) {
+    if (this.props.onEnterStep) {
+      event.preventDefault();
+      this.props.onEnterStep(this.props.route, this.props.stepId);
+    }
+  }
+
+  render() {
+    const { route, screenReaderText, className, children } = this.props;
+    const onClick = event => this.handleClick(event);
+    return (
+      <a href={route} onClick={onClick} className={className}>
+        {children}
+        <span className="ds-u-visibility--screen-reader">
+          {' '}
+          {screenReaderText}
+        </span>
+      </a>
+    );
+  }
+}
+
+StepLink.propTypes = {
+  /**
+   * Label text or HTML.
+   */
+  children: PropTypes.node.isRequired,
+  route: PropTypes.string.isRequired,
+  stepId: PropTypes.string.isRequired,
```

Comment:
What's the id used for? Does it have to be something that's required?

Reply:
Really it should be required in the `stepShape` because it's used as a key in the lists. We know that if it is a required part of the step that it's always going to be passed to this component by the components that use it (because we control the only components that use it, and I purposely forced the package not to expose those sub-components).